var/spack/repos/builtin/packages/amrfinder/package.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class Amrfinder(MakefilePackage):
+    """NCBI AMRFinderPlus: This software and the accompanying database identify
+       acquired antimicrobial resistance genes in bacterial protein and/or
+       assembled nucleotide sequences as well as known resistance-associated
+       point mutations for several taxa. With AMRFinderPlus we added select
+       members of additional classes of genes such as virulence factors,
+       biocide, heat, acid, and metal resistance genes."""
+
+    homepage = "https://github.com/ncbi/amr/wiki"
+    url      = "https://github.com/ncbi/amr/archive/refs/tags/amrfinder_v3.10.30.tar.gz"
+
+    version('3.10.30', sha256='2f1e30b86935a27cee740bd7229a41fbce278f2f60b33b8e51592bab8bdf23f1')
+    version('3.10.24', sha256='fce299c980cda740dcc4f53f9b2dc9061c856213e5bdbc2c339185a5fb7dcf6a')
+
+    depends_on('blast-plus')
+    depends_on('hmmer')
+    depends_on('curl')
+
+    def edit(self, spec, prefix):
+        mkdirp(prefix.bin)
+        mkdirp(prefix.share)
+        env['INSTALL_DIR'] = prefix.bin
+        env['DEFAULT_DB_DIR'] = prefix.share
```

Comment:
Last minor request.We are overriding the `edit` function, but we are effectively not doing any editing. I think:
```suggestion
    def setup_build_environment(self, env):
        env.set('INSTALL_DIR', prefix.bin)
        env.set('DEFAULT_DB_DIR', prefix.share)

    @run_before('build')
    def create_bin_and_share(self):
        mkdirp(self.spec.prefix.bin)
        mkdirp(self.spec.prefix.share)
```
Does this work too?

Reply:
Confirmed that this works and the database downloads to the correct location